### PR TITLE
Add model tests

### LINKED ISSUE #105: Add basic model unit tests for associations & validations
Let's expand our test coverage as much as we think we can using shoulda following the https://github.com/rubyforgood/pet-rescue/pull/102

Don't focus too heavily on being 100% covered, just the basics like associations / validation

### PR DESCRIPTION
**Pull request by Thrillberg**
Fixes #105

### Describe your change in plain English.
I've added some model tests on validations and relationships. I skipped models that I think we're going to remove like the `*_account` models.

### Link to the issue
#105

### CODE REVIEW

File: app/models/match.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Match < ApplicationRecord
   belongs_to :adopter_account
   belongs_to :organization
 
-  validate :belongs_to_same_organization_as_pet
+  validate :belongs_to_same_organization_as_pet, if: ->{ pet.present? }
```
Comment (Thrillberg) [inline app/models/match.rb:28]: This is a bugfix! Thanks testing (and @edwinthinks)

**Review by edwinthinks — APPROVED** [review]
